xdsl/dialects/riscv_debug.py

```diff
@@ -0,0 +1,76 @@
+from collections.abc import Sequence, Set
+
+from xdsl.dialects import riscv
+from xdsl.dialects.builtin import StringAttr
+from xdsl.ir import Attribute, Dialect, SSAValue
+from xdsl.irdl import IRDLOperation, attr_def, irdl_op_definition, var_operand_def
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+
+
+@irdl_op_definition
+class PrintfOp(IRDLOperation, riscv.RISCVInstruction):
+    """
+    An instruction to print the contents of registers when emulating riscv code.
+
+    Is not a real instruction in the RISC-V instruction set, but supported by riscemu
+    and xDSL's interpreter.
+
+    During assembly emission, the results are printed before the operands:
+
+    ``` python
+    s0 = riscv.GetRegisterOp(Registers.s0).res
+    s1 = riscv.GetRegisterOp(Registers.s1).res
+    op = PrintfOp("s0: {}, s1: {}", (s0, s1))
+
+    op.assembly_line()   # "printf "s0: {}, s1: {}", s0, s1"
```

Comment:
This line seems to have a few too many quotation marks?

Reply:
ah it's just the wrong quotation marks